api/router.go

```diff
@@ -51,6 +51,21 @@ func Router(c *ctx.AptlyContext) http.Handler {
 		MetricsCollectorRegistrar.Register(router)
 	}
 
+	if c.Config().ServeInAPIMode {
```

Comment:
Cool stuff! What do you think if the `/repos/` endpoint would return  a list (as json?) of all 
the published repos?

Reply:
I think, if we really want to extend changes scope to that point, it should be not a json response, but a "directory listing"-like html response similar to that generated by `c.FileFromFS` when requested path is a directory.

Reply:
How would I determine which repos have been published?

Reply:
@mmianl something like that: https://github.com/aptly-dev/aptly/blob/77e02bf7a363c78f9c82792a569c8a5a496c1f06/cmd/publish_list.go#L85-L96

But we need to filter out only `filesystem` publishes, if I get the idea right.

Reply:
Agreed with @aol-nnov , directory listing response like would be nice!
But @mmianl if you think this is out of scope I am happy to merge as is/

Reply:
@randombenj I could fiddle with it in a separate PR if my time permits and/or @mmianl won't come with a solution in this PR.

Anyway, current changeset provides viable support for simultaneous serving of files and API and could be merged as is.

Reply:
@mmianl @aol-nnov could you provide an example output for this?

Reply:
@mmianl I've left my laptop in the office. I'll try to arrange some example tomorrow

Reply:
@mmianl I've slept with the idea and actually, it is much easier, than listing publishes an filtering them.

All we need is to list publish prefixes. Rest is handled by the `/repos/:storage/*pkgPath` route!

If you grant me write permissions for this PR, I could push a changeset. Meanwhile find a patch for you latest changes:


```patch
diff --git a/api/repos.go b/api/repos.go
index 8a38dbda..11eddb9e 100644
--- a/api/repos.go
+++ b/api/repos.go
@@ -15,30 +15,20 @@ import (
        "github.com/aptly-dev/aptly/task"
        "github.com/aptly-dev/aptly/utils"
        "github.com/gin-gonic/gin"
-       "github.com/rs/zerolog/log"
 )
 
 // GET /repos
-// todo: limit to filesystem repos only
-// todo: change output format
-func reposListInAPIMode(c *gin.Context) {
-       repos := []*deb.PublishedRepo{}
+func reposListInAPIMode(localRepos map[string]utils.FileSystemPublishRoot) gin.HandlerFunc {
 
-       _ = context.NewCollectionFactory().PublishedRepoCollection().ForEach(func(repo *deb.PublishedRepo) error {
-               err := context.NewCollectionFactory().PublishedRepoCollection().LoadComplete(repo, context.NewCollectionFactory())
-               if err != nil {
-                       msg := fmt.Sprintf(
-                               "Error %s found while getting list of published repos (prefix:%s / distribution:%s / component:%s\n).",
-                               err, repo.StoragePrefix(), repo.Distribution, repo.Components())
-                       log.Debug().Msg(msg)
-                       return err
+       return func(c *gin.Context) {
+               res := c.Writer
+               res.WriteString("<html><body><pre>")
+               for publishPrefix := range localRepos {
+                       res.WriteString(fmt.Sprintf("<a href=\"%[1]s/\">%[1]s</a>\n", publishPrefix))
                }
-
-               repos = append(repos, repo)
-               return nil
-       })
-
-       c.JSON(200, repos)
+               res.WriteString("</pre></body></html>")
+               res.Flush()
+       }
 }
 
 // GET /repos/:storage/*pkgPath
diff --git a/api/router.go b/api/router.go
index 1e368192..ea1dc27f 100644
--- a/api/router.go
+++ b/api/router.go
@@ -53,7 +53,7 @@ func Router(c *ctx.AptlyContext) http.Handler {
        }
 
        if c.Config().ServeInAPIMode {
-               router.GET("/repos/", reposListInAPIMode)
+               router.GET("/repos/", reposListInAPIMode(c.Config().FileSystemPublishRoots))
                router.GET("/repos/:storage/*pkgPath", reposServeInAPIMode)
        }
```

Reply:
Updated accordoing to @aol-nnov's feedback. I changed a few details to make the output of `repos` consistent with the `FileFromFS()` output. I also updated the handler for `/repos/:storage/*pkgPath` according to the above suggestion.

@randombenj, @aol-nnov  what do you think?

Reply:
Love it thanks :tada: